util/timer/src/timer.rs

```diff
@@ -790,6 +844,7 @@ mod tests {
         let (called_at, token) = value.unwrap();
         assert_eq!(token, timer_token);
         assert!(similar(called_at, begin + tick())); // called_at = now + ticksufficiently small
+        timer_loop.stop();
```

Comment:
Should we always call stop?

Reply:
I've added `impl Drop for TimerLoop`. `stop()` will clear remaining all schedule and wait for threads to exit, while `drop()` will try to deliver remaining all schedule, and threads will eventually stop. This is similar to `std::thread::JoinHandle`. Explicit `join()` will wait for it to exit, while `drop()` will silently finish its task.